server/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -325,13 +330,25 @@ public void connectToNode(DiscoveryNode node) throws ConnectTransportException {
      * @param connectionProfile the connection profile to use when connecting to this node
      */
     public void connectToNode(final DiscoveryNode node, ConnectionProfile connectionProfile) {
+        connectToNode(node, connectionProfile, true);
+    }
+
+    /**
+     * Connect to the specified node with the given connection profile. This method allows the caller to
+     * specify whether the node connection should be validated against the specified node.
+     *
+     * @param node the node to connect to
+     * @param connectionProfile the connection profile to use when connecting to this node
+     * @param validateNodeConnection boolean indicating if the node connection should be validated
+     */
+    public void connectToNode(final DiscoveryNode node, ConnectionProfile connectionProfile, boolean validateNodeConnection) {
```

Comment:
can we please not add another API here but rather try to detect if we are a client or a node? I mean down the road we don't want to support this anymore so lets make sure we don't have special public APIs. I can imagine a boolean on the ctor or even check if ClusterService is != null (not that intuitive). You can just add an overloaded ctor instead.

Reply:
we can also just look at the settings like this `TransportClient.CLIENT_TYPE.equals(settings.get(Client.CLIENT_TYPE_SETTING_S.getKey()));`